### PR TITLE
ui: add dropdowns for filtering shared workflows

### LINKED ISSUE #367: ui: implement double dropdown design for filtering shared workflows
Implement the following double dropdown design for filtering shared workflows:

RFC #3 Your workflows_ dropdown (1)

### PR DESCRIPTION
**Pull request by DaanRosendal**
Adds the dropdowns for filtering shared workflows like so:
image

The width of the workflow list has been increased so that the dropdowns could fit on the same row as the existing filters/sorters.

Closes #367

### CODE REVIEW

File: reana-ui/src/pages/workflowList/components/WorkflowSharingFilter.module.scss [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  margin-left: -1px;
+}
+
+#selected-user-dropdown > div[aria-atomic="true"] {
+  max-width: 105px;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  height: 14+2px;
+  margin-bottom: -2px;
```
Comment (giuseppe-steduto) [inline reana-ui/src/pages/workflowList/components/WorkflowSharingFilter.module.scss:41]: I think we can remove these lines, correct?

**Review by giuseppe-steduto — COMMENTED** [review]
Good job! Left some comments, mainly cosmetics and about the user fetch actions.

**Comment by mdonadoni** [comment]
Included in https://github.com/reanahub/reana-ui/pull/375

**Comment by mdonadoni** [comment]
Included in https://github.com/reanahub/reana-ui/pull/375

**Pull request closed without merging**